src/unstakeAg/index.ts

```diff
@@ -799,20 +835,53 @@ export class UnstakeAg {
       user,
       feeAccounts,
       assumeAtasExist,
+      asLegacyTransaction,
     });
-    if (!exchangeReturn.setupTransaction) {
-      exchangeReturn.setupTransaction = {
+
+    if (asLegacyTransaction) {
+      const legacyExchangeReturn = exchangeReturn as ExchangeReturn;
+      legacyExchangeReturn.setupTransaction = {
         tx: new Transaction(),
         signers: [],
       };
+      legacyExchangeReturn.setupTransaction.tx.instructions.unshift(
+        ...withdrawStakeInstructions,
+      );
+      if (isNewStakeAccountKeypair(newStakeAccount)) {
+        legacyExchangeReturn.setupTransaction.signers.push(newStakeAccount);
+      }
+      return tryMergeExchangeReturn(user, legacyExchangeReturn);
     }
-    exchangeReturn.setupTransaction.tx.instructions.unshift(
-      ...withdrawStakeInstructions,
+
+    const v0ExchangeReturn = exchangeReturn as ExchangeReturnV0;
+    const addressLookupTableAccounts = await Promise.all(
+      v0ExchangeReturn.unstakeTransaction.message.addressTableLookups.map(
+        async (lookup) =>
+          new AddressLookupTableAccount({
+            key: lookup.accountKey,
+            state: AddressLookupTableAccount.deserialize(
+              await this.connection
+                .getAccountInfo(lookup.accountKey)
+                .then((res) => res!.data),
+            ),
+          }),
+      ),
+    );
```

Comment:
this is pretty annoying having to refetch the lookup table account data from on-chain. I wonder if theres a better way about it. Ideally we add `LUT_PK` to this too and do a `chunkedGetMultipleAccountInfos` and deserialize all the lut accounts in one shot

Reply:
this is from jup docs https://docs.jup.ag/integrating-jupiter/composing-with-versioned-transactions#4.-fetch-the-address-lookup-table-accounts